bin/node/src/flags/globals.rs

```diff
@@ -20,3 +22,18 @@ pub struct GlobalArgs {
     )]
     pub metrics_port: u16,
 }
+
+impl GlobalArgs {
+    /// Returns the signer [`Address`] from the rollup config for the given l2 chain id.
+    pub fn signer(&self) -> anyhow::Result<Address> {
+        let id = self.l2_chain_id;
+        ROLLUP_CONFIGS
+            .get(&id)
+            .ok_or(anyhow::anyhow!("No rollup config found for chain ID: {id}"))?
+            .genesis
+            .system_config
+            .as_ref()
+            .map(|c| c.batcher_address)
+            .ok_or(anyhow::anyhow!("No system config found for chain ID: {id}"))
```

Comment:
Can we do this? The unsafe block signer can be changed in the `SystemConfig`, so it's not guaranteed to be the same as the one in the genesis state of the `SystemConfig`. Base for example changed theirs recently, and that bit us over in kona-proof.

Reply:
#1326